src/utils.rs

```diff
@@ -101,6 +101,10 @@ active boot switches: -d:release\n",
             stdout: String::from("4.08.1\n"),
             stderr: String::default(),
         }),
+        "perl -e printf q#%vd#,$^V;" => Some(CommandOutput {
```

Comment:
```suggestion
        "perl -e 'printf q#%vd#,$^V;'" => Some(CommandOutput {
```

Reply:
With your modifcations tests fails..., just tested on my machine.

![image](https://user-images.githubusercontent.com/4223252/88601633-71f53a80-d071-11ea-849f-f481eb631504.png)

Reply:
Let's see. I started with:
```
starship on  perl-module [$!] is 📦 v0.44.0 via 🦀 v1.44.1
❯ git show --shortstat
commit 481401be552a5c42552e006a7b9bb89dc38e7603 (HEAD -> perl-module)
Author: Stephane Benoit <stefb@wizzz.net>
Date:   Mon Jul 27 20:50:39 2020 +0200

    Fixed missing new line

 1 file changed, 1 insertion(+), 1 deletion(-)
```

Then, I made changes both to mock and actual invocation:
```diff
diff --git a/src/modules/perl.rs b/src/modules/perl.rs
index 08fd824..1854ac2 100644
--- a/src/modules/perl.rs
+++ b/src/modules/perl.rs
@@ -28,7 +28,7 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         return None;
     }
 
-    let perl_version = utils::exec_cmd("perl", &["-e", "printf q#%vd#,$^V;"])?.stdout;
+    let perl_version = utils::exec_cmd("perl", &["-e", "'printf q#%vd#,$^V;'"])?.stdout;
 
     let mut module = context.new_module("perl");
     let config: PerlConfig = PerlConfig::try_load(module.config);
diff --git a/src/utils.rs b/src/utils.rs
index c83bea6..373fec2 100644
--- a/src/utils.rs
+++ b/src/utils.rs
@@ -101,7 +101,7 @@ active boot switches: -d:release\n",
             stdout: String::from("4.08.1\n"),
             stderr: String::default(),
         }),
-        "perl -e printf q#%vd#,$^V;" => Some(CommandOutput {
+        "perl -e 'printf q#%vd#,$^V;'" => Some(CommandOutput {
             stdout: String::from("5.26.1"),
             stderr: String::default(),
         }),

```

For the record, I'm having Perl 5.32.0 installed locally:
```
starship on  perl-module [$!] is 📦 v0.44.0 via 🦀 v1.44.1
❯ perl --version

This is perl 5, version 32, subversion 0 (v5.32.0) built for darwin-thread-multi-2level

Copyright 1987-2020, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.
```

I've run the test:
```
starship on  perl-module [$!] is 📦 v0.44.0 via 🦀 v1.44.1 took 1m6s
❯ cargo test modules::perl::folder_with_perl_files
warning: method is never used: `set_value`
  --> src/segment.rs:46:5
   |
46 | /     pub fn set_value<T>(&mut self, value: T) -> &mut Self
47 | |     where
48 | |         T: Into<String>,
49 | |     {
50 | |         self.value = value.into();
51 | |         self
52 | |     }
   | |_____^
   |
   = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted

warning: method is never used: `is_empty`
   --> src/module.rs:101:5
    |
101 |     pub fn is_empty(&self) -> bool {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: method is never used: `new`
  --> src/segment.rs:21:5
   |
21 |     pub fn new(name: &str) -> Self {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `set_value`
  --> src/segment.rs:46:5
   |
46 | /     pub fn set_value<T>(&mut self, value: T) -> &mut Self
47 | |     where
48 | |         T: Into<String>,
49 | |     {
50 | |         self.value = value.into();
51 | |         self
52 | |     }
   | |_____^

warning: method is never used: `is_empty`
  --> src/segment.rs:68:5
   |
68 |     pub fn is_empty(&self) -> bool {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 4 warnings emitted

    Finished test [unoptimized + debuginfo] target(s) in 0.74s
     Running target/debug/deps/starship-2b44a645abddee21

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 251 filtered out

     Running target/debug/deps/starship-b3862c2e5a1d1776

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 263 filtered out

     Running target/debug/deps/testsuite-d9f5f469e468477c

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 181 filtered out


```

And it passed ✅

Reply:
@stefb69 It seems that the key to success is to quote on both sides (invocation and mock). Let me know if that works for you 🤞

Reply:
I see your point but please try the resulting binary in a perl folder : 
![image](https://user-images.githubusercontent.com/4223252/88604336-876d6300-d077-11ea-9a09-fdaf1175a52e.png)
Same as last time, i already tried and had the same issue. The only way everything works ok is without quotes

Reply:
And actually the tests do not fail in the automated tests, it is another module that fails with ubuntu nightly and macos nightly

Reply:
![image](https://user-images.githubusercontent.com/4223252/88605126-5aba4b00-d079-11ea-8ec1-21f42555c7c8.png)

Reply:
> And actually the tests do not fail in the automated tests, it is another module that fails with ubuntu nightly and macos nightly

You can ignore those, they fail sporadically on other PRs too...

Reply:
> I see your point but please try the resulting binary in a perl folder :
> ![image](https://user-images.githubusercontent.com/4223252/88604336-876d6300-d077-11ea-9a09-fdaf1175a52e.png)
> Same as last time, i already tried and had the same issue. The only way everything works ok is without quotes

Wow, you are right. This is interesting 🤔

Reply:
Well, what can I say? I was wrong about this and sorry for pushing you into reiterating on what should've stayed closed/resolved topic 🤦 
Actually this provided me with a valuable lesson 👍  and now I need to go and revert portions of #1491 (like this for instance https://github.com/starship/starship/pull/1491/files#diff-3dbd16223f25e8b5e0bcf099743107e9R104)